### ISSUE TITLE
zim: links do not follow zim link behaviour (both relative and absolute paths)

### ISSUE THREAD

**Issue by Cratesmith** [gsantner/markor#1223]
#### General information

* **App version: 2.5.0** 
* **System: Android 9** 

#### Description
Zim links don't follow the correct search behavior (as specified https://zim-wiki.org/manual/Help/Links.html) resulting in incorrect navigation when loading a zim notebook.

1. Relative links
Relative paths should search upward from the parent to resolve as much of the path as possible before creating a new page, currently it seems like no search occurs. 

For example in the page Book:Chapter1:Page3 a link to [[Chapter1:Page2]] should resolve to  Book:Chapter1:Page2, instead of Book:Chapter1:Chapter1:Page2 as it currently does.

2. Absolute links
Links starting with ':' are supposed to be top-level links, relative to the top most folder of the notebook. Currently ':' seems to literally translate to '/' which results in documents being placed in strange places outside the Markor documents directory. 

This should probably resolve by finding the parent folder that contains the 'notebook.zim' file, and if none is found default to the Markor documents directory.

These should resolve to the top level of the notebook (whichever folder is designated as the Makor top level folder). 

3. Child links 
Links starting with '+' are supposed to be resolved using the current page as a parent. For instance in Book:Chapter1:Page3 a link to [[+Figure1]] should resolve to Book:Chapter1:Page3:Figure1. Presently it links somewhere else... I'm not entirely sure where. There are some odd placed text files on my device as a result of this.

For now it might be a good idea to disable linking until these issues are resolved? As the current links if you open up an existing notebook will almost always be incompatible with the zim desktop app.

**Comment by gsantner** [comment]
I don't use Zim, but is there really a need for that much magic for referencing files? Can you not just reference it by filename or soemthing? my humble opinion.

**Comment by Cratesmith** [comment]
Well... the use case for zim format in Markor is letting people view/edit their notebooks on the go. Something that's been sorely missing till now. As it stands, Markor isn't supporting the most important part of the zim format.

Being compatibble with zim notebooks made in zim is key, as essentially all zim content is made in that and it automatically makes these relative links, which will all be incorrect in Markor (resulting in potentially creating files in very odd places!)

As for the logic, it's actually not that complicated. You just need to process the link between. 
1. starts with +: prepend children folder of current document to path
2. starts with ':' : prepend notebook root to path
3. starts with '?' : ignore (unless you want to support interwiki)
4. any others: do a parent resolve up to the document root, create as peer to current document as fallback.
Then just replace ':' with '/' as is already being done and it's sorted.

**Comment by gsantner** [comment]
You seem to know Zim pretty well and  being a active user of it. I do neither, so I would be happy if you can help and improve Markor. Make changes and pull request, thanks.

**Comment by Cratesmith** [comment]
I'll give it a shot when I've got some time. 

I'd assumed that somene in the Markor community was already spearheading this from being a zim user but hadn't tested a full zim notebook and found this problem.

**Comment by gsantner** [comment]
What is currently there is the first & initial version of Zim support. Yes it's 1-2 interested Zim users, still their time & interest of spending time is limited. So there are more people that want things than there are people+time that do them.

**Comment by fredericjacob** [comment]
@Cratesmith I implemented most of the Zim features and yes, interwiki links are not yet implemented (apart from very simple links to files in the current location), I'm aware of this. They were not top priority compared with the other features and personally, I don't consider them THAT important although I have a quite large notebook with more than 500 pages. But this depends on the usage, I guess.
If you want to join and participate in the Markor Zim development to implement this feature - great, I'd really appreciate that! If you need some help with the Zim-related code, just tell me. :)

**Comment by gsantner** [comment]
@Cratesmith Did you still use Markor? Want to improve it? Help on Zim support would be totally welcome as nobody is actively working on it.

**Comment by Cratesmith** [comment]
Yep! I use it mostly in a read only capacity, or to edit individual pages
on the go.
That might be a bit harder soon as I'm using google drive to sync my zim
notebooks to my phone and that's going away soon but... I'll figure out
something.

I'd like to help out but at present I'm far too overloaded both
professionally and personally to offer any time to the project.

**Comment by fredericjacob** [comment]
@gsantner You can assign this issue to me, I started working on the feature to follow zim wiki links. I will create a work-in-progress PR on this soon.

@Cratesmith I can recommend Syncthing if you are searching for an alternative to Google Drive, works really well for me - but there are of course other options as well.
